test/class-test.js

```diff
@@ -68,3 +74,45 @@ describe('static server', () => {
     });
 });
 
+describe('Download and upload', () => {
+    const fileName = 'imagetest.jpg';
+    const filePath = './test/data/' + fileName;
+    const newPath  = './resources/upload/' + fileName;
+
+    afterEach(async () => {
+        await finishAll();
+    });
+
+    after(async () => {
+        await fs.unlink(newPath);
+    });
+
+    it('The file should upload successfully', async () => {
+        const server = await startServer(1337);
+
+        await request(server.app)
+            .post('/upload')
+            .attach('image', filePath)
+            .expect(200);
+
+        const res = await fs.stat(newPath);
+
+        assert(res.isFile());
+    });
+
+    it('The file should download successfully', async () => {
+        const server = await startServer(1337);
+        const buf = await fs.readFile('./resources/files/picture.jpg', (err, buffer) => {
```

Comment:
I think we can simplify it. `readFile` returns Promise. It means that the following code should work.
```js
const buf = await fs.readFile('./resources/files/picture.jpg');
```